tools/decoder.py

```diff
@@ -181,6 +189,9 @@ def select_tool(toolchain_path, tool, func):
     path = f"xtensa-lx106-elf-{tool}"
     if toolchain_path:
         path = os.path.join(toolchain_path, path)
+        if not os.path.isfile(path):
+            print(f"error: not existing tool (path -- tool): '{path}'", file=sys.stderr)
+            sys.exit(1)
```

Comment:
Already handled when trying to subprocess.Popen(path, ...)?
For both cases, with or without toolchain path.

Reply:
Agreed.
This one checks on script startup and helps user to build the command line up by not failing later when the first stack dump occurs.

Reply:
https://docs.python.org/3/library/shutil.html#shutil.which
e.g.
```python
if not shutil.which(path):
    raise FileNotFoundError(path)
```